src/App.tsx

```diff
@@ -1,14 +1,69 @@
 /* eslint-disable max-len */
-import React from 'react';
+import {
+  useEffect,
+  useState,
+  FC,
+  ChangeEvent,
+  useCallback,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { CompletedStatus } from './types/CompletedStatus';
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [selectedStatus, setSelectedStatus] = useState<CompletedStatus>(CompletedStatus.All);
+  const [clickedTodoId, setClickedTodoId] = useState(0);
+  const [userTodo, setUserTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos().then(result => {
+      setTodos(result);
+    });
+  }, []);
+
+  const handleInput = useCallback((event: ChangeEvent<HTMLInputElement> | string) => {
+    if (typeof event !== 'string') {
+      setQuery(event.target.value);
+    }
```

Comment:
```suggestion
  const handleChange = useCallback((event: ChangeEvent<HTMLInputElement>) => {
      setQuery(event.target.value);
```

Reply:
![image](https://github.com/mate-academy/react_dynamic-list-of-todos/assets/49568391/dc889ccb-12e8-4a1d-9901-eb3d079de7d9)